src/components/state.tsx

```diff
@@ -0,0 +1,31 @@
+import { createContext, ReactNode, useState } from 'react';
+import { Person } from '../types';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const MyContext = createContext<any>(null);
+
+interface MyProvideProps {
+  children: ReactNode;
+}
+
+export const MyProvider = ({ children }: MyProvideProps) => {
+  const [person, setPerson] = useState<Person[]>([]);
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [selectedPerson, setSelectedPerson] = useState<string | null>(null);
```

Comment:
For `selectedPerson`, consider initializing the state with `null` without the explicit type string. TypeScript can infer the type as `string | null` based on the initial value.